### PR TITLE
[Untested] Increases fort movement efficiency

### PR DESCRIPTION
**Pull request by Cameronlund4**
Makes it so distance tracking is done after every fort, instead of just
at the beginning. This should make walking times between pokestops much
less in certain cases. (For example, on the old loop, if fort a was
2.5mi north, fort b was 2.7 mi north, and fort c was 2.6mi south, the
program would go a->c->b, which is insanely inefficient)

Note, unsure of the change on line 116. Without that change the current
code will get in an infinite loop once it goes through all forts it
knows, but might undo whatever that line was trying to do. Instead of
breaking, it just moves on to the next fort.

### CODE REVIEW

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -100,17 +100,20 @@ def work_on_cell(self, cell, position, include_fort_on_path):
...
-                forts.sort(key=lambda x: distance(self.position[
-                           0], self.position[1], x['latitude'], x['longitude']))
-                for fort in cell['forts']:
+           
+                while len(forts) > 0:
+                    # Sort all by distance from current pos- eventually this should
+                    # build graph & A* it
+                    forts.sort(key=lambda x: distance(self.position[
+                               0], self.position[1], x['latitude'], x['longitude']))
+                    fort = forts[0]
```
Comment (solderzzc) [inline pokemongo_bot/__init__.py]: Did it exclude the current one?

**Comment by Cameronlund4** [comment]
@solderzzc If you mean does it replace the old loop, then yes. Otherwise not sure what you mean.

**Comment by solderzzc** [comment]
I mean we need delete the current position from list, is that correct?

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -110,17 +110,20 @@ def work_on_cell(self, cell, position, include_fort_on_path):
...
                     worker = SeenFortWorker(fort, self)
                     hack_chain = worker.work()
```
Comment (ChronoAS) [inline pokemongo_bot/__init__.py]: Shouldn't you set hack_chain to a value after you remove a fort? Like moving a like 121 to line 124

Comment (solderzzc) [inline pokemongo_bot/__init__.py]: NO, hack_chain is the record on server side , mean server know how frequently you search fort.

File: pokemongo_bot/__init__.py [hunk 3]
```diff
@@ -110,17 +110,20 @@ def work_on_cell(self, cell, position, include_fort_on_path):
...
-                    if hack_chain > 10:
+                    if fort in forts:
+                        fort.remove(fort)
```
Comment (ChronoAS) [inline pokemongo_bot/__init__.py]: forts.remove(fort)

then

**Comment by solderzzc** [comment]
You should test it first them merge.

**Comment by solderzzc** [comment]
cleaning up the pull-request folder, please reopen when you done the rebase.

**Pull request closed without merging**